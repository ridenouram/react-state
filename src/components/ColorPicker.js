import React from 'react';
import PropTypes from 'prop-types';

export default function ColorPicker({ colorFunction }) {
  return (
    <>
      <button style={{ backgroundColor: 'red', width: '75px', height: '50px' }} onClick={() => colorFunction('red')}>Red</button>
      <button style={{ backgroundColor: 'yellow', width: '75px', height: '50px' }} onClick={() => colorFunction('yellow')}>yellow</button>
      <button style={{ backgroundColor: 'blue', width: '75px', height: '50px' }} onClick={() => colorFunction('blue')}>blue</button>
    </>
  );
}

ColorPicker.propTypes = {
  colorFunction: PropTypes.func.isRequired
};
