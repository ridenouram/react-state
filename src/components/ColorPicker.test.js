import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('colorpicker component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ColorPicker colorFunction={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('selects red on button click', () => {
    const mock = jest.fn();
    const wrapper = shallow(<ColorPicker colorFunction={mock} />);
    wrapper.find('button').at(0).simulate('click');
    expect(mock).toHaveBeenCalledWith('red');
  });

  it('selects yellow on button click', () => {
    const mock = jest.fn();
    const wrapper = shallow(<ColorPicker colorFunction={mock} />);
    wrapper.find('button').at(1).simulate('click');
    expect(mock).toHaveBeenCalledWith('yellow');
  });

  it('selects blue on button click', () => {
    const mock = jest.fn();
    const wrapper = shallow(<ColorPicker colorFunction={mock} />);
    wrapper.find('button').at(2).simulate('click');
    expect(mock).toHaveBeenCalledWith('blue');
  });
});

