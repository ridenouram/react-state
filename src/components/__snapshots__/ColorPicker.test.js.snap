// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorpicker component matches snapshot 1`] = `
<Fragment>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "height": "50px",
        "width": "75px",
      }
    }
  >
    Red
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "height": "50px",
        "width": "75px",
      }
    }
  >
    yellow
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "height": "50px",
        "width": "75px",
      }
    }
  >
    blue
  </button>
</Fragment>
`;
